### PR TITLE
Adapting the code for python 3

### PR DESCRIPTION
**Pull request by ikvision**
I have edited the code to be python 3 compatible, I tested the code according to ReadeMe.md

### CODE REVIEW

File: data/groups.py [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
+        if not (count % 10000): print(count)
         brand = row[1]
         if brand in brands: brands[brand] += 1
         else: brands[brand] = 1
         category = row[3].split(' / ')[0]
         if category in categories: categories[category] += 1
         else: categories[category] = 1
-    print sorted(brands.items(), key=itemgetter(1))
-    print sorted(categories.items(), key=itemgetter(1))
+    print( sorted(brands.items(), key=itemgetter(1)))
```
Comment (etano) [inline data/groups.py]: space

File: data/normalize.py [hunk 2]
```diff
@@ -1,27 +1,43 @@
...
+
+# def unescape(s):
+#     p = HTMLParser()
+#     # p.save_bgn()
+#     try:
+#         p.feed(str(s))
+#     except:
+#         return s
+#     p.close()
+#     return p.unescape(str(s))
```
Comment (etano) [inline data/normalize.py]: just remove, don't comment out

File: extract.py [hunk 3]
```diff
@@ -25,18 +25,18 @@ def process(row, tokenizer, classifier, ner):
...
 
     # Extract entities
     data = tokenizer.tokenize([row['name']])
     tags = ner.tag(data)[0]
     brand, brand_started = '', False
     for word, tag in zip(row['name'].split(' '), tags):
-        max_tag = max(tag.items(), key=itemgetter(1))[0]
-        if max_tag == u'B-B' and (not brand_started):
+        max_tag = max(list(tag.items()), key=itemgetter(1))[0]
+        if max_tag == 'B-B' and (not brand_started):
```
Comment (etano) [inline extract.py]: probably safer to do `if 'B-B' in max_tag`

File: extract.py [hunk 4]
```diff
@@ -25,18 +25,18 @@ def process(row, tokenizer, classifier, ner):
...
             brand = word
             brand_started = True
-        elif max_tag == u'I-B' and brand_started:
+        elif max_tag == 'I-B' and brand_started:
```
Comment (etano) [inline extract.py]: same as above

File: ner.py [hunk 5]
```diff
@@ -202,7 +202,8 @@ def evaluate(self, x_test, y_test, batch_size=256):
         predictions_last_epoch = self.model.predict(x_test, batch_size=batch_size, verbose=1)
         predicted_classes = np.argmax(predictions_last_epoch, axis=2).flatten()
         y_val = np.argmax(y_test, axis=2).flatten()
-        target_names = ['']*(max(self.tag_map.itervalues())+1)
+        target_names = ['']*(max(self.tag_map.values())+1)
         for category in self.tag_map:
             target_names[self.tag_map[category]] = category
-        print(classification_report(y_val, predicted_classes, target_names=target_names, digits = 6))
+
+        print((classification_report(y_val, predicted_classes, target_names=target_names, digits = 6,labels=range(len(target_names)))))
```
Comment (etano) [inline ner.py]: space before `labels`

File: tokenizer.py [hunk 6]
```diff
@@ -2,7 +2,7 @@
 
 import os
 import numpy as np
-import cPickle
+import pickle
```
Comment (etano) [inline tokenizer.py]: use a try/except to `import cPickle as pickle` b/c it really is a lot faster.

File: train_classifier.py [hunk 7]
```diff
@@ -43,7 +43,7 @@ def main(argv):
 
     # Compile classifier network and train
     classifier.compile(tokenizer)
-    classifier.train(data, labels)
+    classifier.train(data, labels,epochs=2)
```
Comment (etano) [inline train_classifier.py]: space before epochs

File: train_ner.py [hunk 8]
```diff
@@ -43,7 +43,7 @@ def main(argv):
 
     # Compile NER network and train
     ner.compile(tokenizer)
-    ner.train(data, labels)
+    ner.train(data, labels,epochs=2)
```
Comment (etano) [inline train_ner.py]: same as above

**Comment by ikvision** [comment]
I appreciate your detailed remarks
I created a new PR with the requested changes

**Pull request merged**